packages/flutter_tools/lib/src/commands/screenshot.dart

```diff
@@ -66,8 +66,8 @@ class ScreenshotCommand extends FlutterCommand {
   @override
   Future<FlutterCommandResult> verifyThenRunCommand(String commandPath) async {
     device = await findTargetDevice();
-    if (device == null) {
-      throwToolExit('Must have a connected device');
+    if (argResults[_kType] == _kDeviceType && device == null) {
```

Comment:
Merge this and and the `device.supportsScreenshot` check into one `if (argResults[_kType] == _kDeviceType)` block

Reply:
The error message will be different in both the cases, no?

1. The first case is where we must have a device connected.
2. The device doesn't support screenshots even when connected.

Were you proposing nesting the `if` conditions inside the parent `if`?

Reply:
Yes

Reply:
Done.